### PR TITLE
New vim docs structure

### PR DESCRIPTION
**Pull request by ii14**
@hrsh7th @elianiva I restructured the vim docs, what do you think about it?

### CODE REVIEW

**Comment by ii14** [comment]
By the way, I don't think README should be just a duplicate of the vim docs.

The docs should explain everything in detail and the README should explain just the general concept behind the plugin and a couple of examples to get you started.

File: doc/compe.txt [hunk 1]
```diff
@@ -184,15 +345,13 @@ VIMSCRIPT CONFIG                                           *compe-usage-ex-viml*
   let g:compe.source.buffer = v:true
   let g:compe.source.calc = v:true
   let g:compe.source.vsnip = v:true
+  let g:compe.source.lamp = v:true
```
Comment (ii14) [inline doc/compe.txt:348]: Should lamp also be in examples, or was it omitted previously on purpose?

Comment (elianiva) [reply doc/compe.txt:348]: yeah, it was omitted on purpose and I think it'll get removed in the future

**Comment by elianiva** [comment]
> By the way, I don't think README should be just a duplicate of the vim docs.
> 
> The docs should explain everything in detail and the README should explain just the general concept behind the plugin and a couple of examples to get you started.

Totally agree, I was copying the readme as a "temporary" solution. Glad you made this!

File: doc/compe.txt [hunk 2]
```diff
@@ -210,93 +369,51 @@ LUA CONFIG                                                  *compe-usage-ex-lua*
       buffer = true;
       calc = true;
       vsnip = true;
+      lamp = true;
       nvim_lsp = true;
```
Comment (elianiva) [inline doc/compe.txt:373]: I think we also have `tags` source.

**Comment by elianiva** [comment]
while you're at it, could you add snippets.nvim link in for the readme (builtin source > external plugin)? I forgot to add it the last time I changed it :laughing:

**Comment by ii14** [comment]
From what I can see, snippets.nvim is already there :)

**Comment by elianiva** [comment]
yeah but it's not linked lile the other ones

File: doc/compe.txt [hunk 3]
```diff
@@ -56,117 +44,290 @@ FEATURES                                                        *compe-features*
...
+Neovim-specific: ~
+nvim_lsp            Neovim's builtin LSP completion.
+nvim_lua            Neovim's Lua "stdlib" completion.
+
+External plugins: ~
+Completion for external plugins.
+Make sure you have the corresponding plugin installed.
+
+vim_lsp             vim-lsp completion.
+lamp                vim-lamp completion.
```
Comment (hrsh7th) [inline doc/compe.txt:319]: Sorry. Please remove it :)

File: doc/compe.txt [hunk 4]
```diff
@@ -212,91 +371,49 @@ LUA CONFIG                                                  *compe-usage-ex-lua*
...
+    To stop the competion process, call `args.abort()`.
 
-- menu
-  Specify item's menu (see `:help complete-items`)
+documentation ~
+    Optional.
+    You can provide documentation for selected items.
+    See nvim_lsp or snippets.nvim source for examples.
 
+confirm ~
```
Comment (hrsh7th) [inline doc/compe.txt:410]: Please add `resolve` for it.

**Comment by hrsh7th** [comment]
Thank you for your much work!

File: doc/compe.txt (hunk 4 above)
Comment (ii14) [reply doc/compe.txt:410]: What `resolve` does? It wasn't in documentation before

Comment (ii14) [reply doc/compe.txt:409]: Looked up the LSP specs and still not entirely sure, but can it be something like this? At least for now
```
resolve
    Optional.
    Resolve completion item.
```

Comment (hrsh7th) [reply doc/compe.txt:409]: Sorry. I had only added it to the wiki page.
https://github.com/hrsh7th/nvim-compe/wiki#resolve-optional

Comment (hrsh7th) [reply doc/compe.txt:415]: Thank you!

**Comment by hrsh7th** [comment]
It looks very good!

Is it ready to merge? @ii14

**Comment by ii14** [comment]
Yes. It's good enough for now :+1:

**Comment by hrsh7th** [comment]
Thank you~~~!

**Pull request merged**